qa/tasks/cephadm.py

```diff
@@ -1962,27 +1962,37 @@ def _samba_ad_dc_conf(ctx, remote, cengine):
     ]
 
 
-@contextlib.contextmanager
-def configure_samba_client_container(ctx, config):
-    # TODO: deduplicate logic between this task and deploy_samba_ad_dc
+def _discover_container_engine(ctx, config):
     role = config.get('role')
-    samba_client_image = config.get(
-        'samba_client_image', 'quay.io/samba.org/samba-client:latest'
-    )
     if not role:
-        raise ConfigError(
-            "you must specify a role to discover container engine / pull image"
-        )
+        raise ConfigError('must specify a role to discover container engine')
     (remote,) = ctx.cluster.only(role).remotes.keys()
     cengine = 'podman'
     try:
-        log.info("Testing if podman is available")
+        log.info('Testing if podman is available')
         remote.run(args=['sudo', cengine, '--help'])
     except CommandFailedError:
-        log.info("Failed to find podman. Using docker")
+        log.info('Failed to find podman. Using docker')
         cengine = 'docker'
+    return cengine
 
-    remote.run(args=['sudo', cengine, 'pull', samba_client_image])
+
+def _pull_samba_container_image(ctx, config, cengine, image):
```

Comment:
Given that this function pulls any image, we could consider dropping samba from it's name.